[IP-942]: Add bic+remittance fields to simplify banktransfers

Verony-makesIT opened the pull request:
## Description
To make payments (for QR code or bank transfers) **to non-SEPA countries**, the use of the **BIC (SWIFT) code is mandatory**.
On a (European) bank transfer or QR code you can also include a **remittance information** description or payment reference indicating which invoice you must/want to pay.

## Related Issue
Feature Request #942 "_Add BIC code and remittance information fields to user table to simplify bank transfers_"
See my comments in Feature Request #942
Accompanying threads on slack: 
- https://invoiceplane.slack.com/archives/C02DQ4VN5/p1691829759284489
- https://invoiceplane.slack.com/archives/C02DQ4VN5/p1692026638514429

## Motivation and Context
To use a QR code (EU EPC) or other electronic invoice formats in the future, BIC is a mandatory field in most cases. 
As you can see in the screenshot below, I had to additionally modify the default public web template by:
1. Adding QRcode.js files to the core/js/ folder.
2. Added some php code (Pay Now button, modal and data) in my template 

## Screenshots (if appropriate):
1. The users **BIC and Remittance Invoice Template tag(s)**
![image](https://github.com/InvoicePlane/InvoicePlane/assets/30659226/b17a6025-1fe6-4c76-89c4-0a2bc0d7eda7)

2. **The result:**
![image](https://github.com/InvoicePlane/InvoicePlane/assets/30659226/8a0cd7ec-b379-4a7b-909a-ee05183a2a86)

## Pull Request Checklist

  * [X] My code follows the code formatting guidelines.
  * [X] I have an issue ID for this pull request.
  * [X] I selected the corresponding branch.
  * [X] I have rebased my changes on top of the corresponding branch.

## Issue Type (Please check one or more)

  * [ ] Bugfix
  * [ ] Improvement of an existing Feature
  * [X] New Feature

nielsdrost7 commented:
Ok, to start, initial code looks good, but I have a question:

> In your screenshot you have a button "Pay with PR code" and a modal with the QR code.
Where in the code from your PR is that modal generated and is the QR code generated?

Verony-makesIT commented:
@nielsdrost7: 
As I mentioned in my PR in the "Motivation and Context" section, I set this up only as **a demo** and to prove that adding BIC and remittance fields are necessary, _as an example_, to deploy QR code (and also e-invoicing). 
In the **assets/core/js/ folder** I placed the **_QR code (.js) files_** and in my **public web template** file, I added **_the code for the modals for the "Pay Now" and "Preview PDF" buttons_**. 
To display the QR code in the modal, I also added some javascript in my **public web template** where I defined the **_invoice data_** and **_QR code variables_**. 
> That way, nothing needs to be changed anywhere in the IP application code itself.

OPGRBILLINGDEV commented:
If BIC payment is instruction for bank to pay via SWIFT, for non-SEPA, OUR/SHA option should be added. So that bank knows, who is paying payment cost (of bank clearance) of bank transfer. Sender of payment (OUR) or in SHA ("acronym of shared") case if it's shared equally among both participants in the payment transfer. If this is not provided payment might end up in dead-end since it's missiong certain options, from usual 102 swift payment form.

Another option would be - if payment is "Same day", or "Express" or not.
